libs/core/schedulers/include/hpx/schedulers/thread_queue.hpp

```diff
@@ -188,7 +189,7 @@ namespace hpx { namespace threads { namespace policies {
                     p = threads::thread_data_stackful::create(
                         data, this, stacksize);
                 }
-                thrd = thread_id_type(p);
+                thrd = thread_id_type(p, false);    // do not addref
```

Comment:
Since we've talked about adding and not adding references before, how about an `enum class addref { yes, no }` or something similar in place of that bool?

Reply:
Good thinking!